**Finalized Queries**

In [2]:
import pandas as pd
import sqlalchemy as sq
print(pd.__version__)
print(sq.__version__)

2.1.0
2.0.36


In [10]:
engine = sq.create_engine('mysql+mysqlconnector://student:NOzLMh4i1gyh7@localhost/student')

In [11]:
df = pd.read_csv("masterdata_with_totalpopulation.csv")
df

CSD  Period                Key  Total population  \
0     Acadia No. 34    2001  Acadia No. 342001               401   
1     Acadia No. 34    2002  Acadia No. 342002               407   
2     Acadia No. 34    2003  Acadia No. 342003               408   
3     Acadia No. 34    2004  Acadia No. 342004               408   
4     Acadia No. 34    2005  Acadia No. 342005               407   
...             ...     ...                ...               ...   
9581     Youngstown    2019     Youngstown2019                84   
9582     Youngstown    2020     Youngstown2020                89   
9583     Youngstown    2021     Youngstown2021               110   
9584     Youngstown    2022     Youngstown2022               109   
9585     Youngstown    2023     Youngstown2023               105   

      Life Expectancy (Years)  Oil Wells (Count)  \
0                         NaN                NaN   
1                         NaN                NaN   
2                         NaN               38.0   
3                         NaN               29.0   
4                         NaN               46.0   
...                       ...                ...   
9581                81.160000                NaN   
9582                82.203333                NaN   
9583                81.923333                NaN   
9584                81.783333                NaN   
9585                      NaN                NaN   

      Natural Gas Production (m^3)  CH4 Equivalent (t)  CO2 Equivalent (t)  \
0                              NaN                 NaN                 NaN   
1                              NaN                 NaN                 NaN   
2                          57081.1                 NaN                 NaN   
3                          95004.2                 0.0                 0.0   
4                         122560.0                 0.0                 0.0   
...                            ...                 ...                 ...   
9581                           NaN                 0.0                 0.0   
9582                           NaN                 0.0                 0.0   
9583                           NaN                 0.0                 0.0   
9584                           NaN                 NaN                 NaN   
9585                           NaN                 NaN                 NaN   

      N2O Equivalent (t)  ...  Movable dwelling (Count)  \
0                    NaN  ...                       0.0   
1                    NaN  ...                       NaN   
2                    NaN  ...                       NaN   
3                    0.0  ...                       NaN   
4                    0.0  ...                       NaN   
...                  ...  ...                       ...   
9581                 0.0  ...                       NaN   
9582                 0.0  ...                       NaN   
9583                 0.0  ...                       0.0   
9584                 NaN  ...                       NaN   
9585                 NaN  ...                       NaN   

      Other attached dwelling (Count)  Row house (Count)  \
0                                 0.0                0.0   
1                                 NaN                NaN   
2                                 NaN                NaN   
3                                 NaN                NaN   
4                                 NaN                NaN   
...                               ...                ...   
9581                              NaN                NaN   
9582                              NaN                NaN   
9583                              0.0                0.0   
9584                              NaN                NaN   
9585                              NaN                NaN   

      Semi-detached house (Count)  Single-detached house (Count)  \
0                             0.0                          160.0   
1                             NaN                            NaN   
2  

**What is the relationship between natural gas production and average AQHI across different CSDs?** 

In [24]:
df_query_1 = pd.read_sql_query('''
    SELECT 
        CSD,
        SUM(`Natural Gas Production (m^3)`) * 100.0 / 
        (SELECT SUM(`Natural Gas Production (m^3)`) FROM masterdata) AS gas_production_percentage,
        AVG(`Adjusted_AQHI`) AS average_aqhi
    FROM 
        masterdata 
    GROUP BY 
        CSD
    HAVING 
        AVG(`Adjusted_AQHI`) > 0
    ORDER BY 
        gas_production_percentage DESC
    LIMIT 5;
''', engine)

df_query_1

CSD  gas_production_percentage  average_aqhi
0            Yellowhead County                  15.282469      2.125167
1  Grande Prairie County No. 1                   4.221192      2.261845
2                Newell County                   3.484994      2.077990
3               Brazeau County                   1.994265      2.310336
4                 Wood Buffalo                   0.886241      1.975499

* Regions with higher gas production percentages, such as Yellowhead County (15.28%) and Grande Prairie County No. 1 (4.22%), tend to have lower average AQHI values (2.12 and 2.26, respectively).
* CSDs like Strathcona County and Edmonton have significantly high AQHI values (2.34 and 2.69) despite very low gas production percentages so it seems like natural gas production is not the sole major contributor to the AQI.
* Calgary and Edmonton have negligible gas production percentages (0.036% and 0.003%) but exhibit some of the highest AQHI values.

**What is the total natural gas production, along with the corresponding average air quality health index (AQHI), and total emissions percentage for each year?** 

In [20]:
df_query_2 = pd.read_sql_query('''
    SELECT 
        Period AS year,
        SUM(`Natural Gas Production (m^3)`) AS total_natural_gas_produced_m3,
        AVG(`Adjusted_AQHI`) AS average_aqhi,
        (SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`)) * 100.0 / 
        (SELECT SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`) FROM masterdata) 
        AS total_emissions_percentage
    FROM 
        masterdata 
    GROUP BY 
        Period
    ORDER BY 
        year;
''', engine)

df_query_2

year  total_natural_gas_produced_m3  average_aqhi  \
0   2001                            NaN           NaN   
1   2002                            NaN           NaN   
2   2003                    166667668.8           NaN   
3   2004                    167774285.4           NaN   
4   2005                    168226730.0           NaN   
5   2006                    167462531.0           NaN   
6   2007                    165551523.0           NaN   
7   2008                    157677562.7           NaN   
8   2009                    146670841.6           NaN   
9   2010                    139377911.8           NaN   
10  2011                    133492877.8      2.318883   
11  2012                    125968456.9      2.208154   
12  2013                    126049560.8      2.305793   
13  2014                    131392285.9      2.235873   
14  2015                    129375243.8      2.196267   
15  2016                    128932304.3      1.971832   
16  2017                    132566858.9      2.216274   
17  2018                    133793900.7      2.429098   
18  2019                    130572454.1      2.166351   
19  2020                    125433571.3      2.105682   
20  2021                    131664222.0      2.283161   
21  2022                    136110700.5      2.269523   
22  2023                    134655966.2           NaN   

    total_emissions_percentage  
0                          NaN  
1                          NaN  
2                          NaN  
3                     4.436803  
4                     4.449358  
5                     4.739095  
6                     4.693722  
7                     4.600118  
8                     4.923258  
9                     5.213325  
10                    5.265267  
11                    5.390878  
12                    5.629673  
13                    5.914261  
14                    5.992075  
15                    5.975827  
16                    6.733430  
17                    6.668335  
18                    6.676714  
19                    6.284277  
20                    6.413584  
21                         NaN  
22                         NaN

* The steady decline in natural gas production does not strongly correlate with AQHI or emissions, indicating other influencing factors such as industrial activity or transportation emissions.
* The emissions percentage increase suggests environmental impacts are not fully mitigated by reduced production.
* Missing data from 2001 and 2002.

**How does the contribution of each CSD to total emissions relate to its average AQHI?** 

In [21]:
df_query_3 = pd.read_sql_query('''
    SELECT 
        CSD,
        (SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`)) * 100.0 / 
        (SELECT SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`) FROM masterdata) 
        AS contribution_percentage,
        AVG(`Adjusted_AQHI`) AS average_aqhi
    FROM 
        masterdata 
    GROUP BY 
        CSD
    ORDER BY 
        contribution_percentage DESC
    LIMIT 5;
''', engine)

df_query_3

CSD  contribution_percentage  average_aqhi
0       Wood Buffalo                29.485486      1.975499
1    Parkland County                14.510808      2.061233
2       Leduc County                 6.898936      1.856974
3  Strathcona County                 6.509499      2.336999
4  Bonnyville No. 87                 6.019841      1.970916

* The top contributors to total emissions, such as Wood Buffalo (29.49%) and Parkland County (14.51%), tend to have relatively low average AQHI values (1.98 and 2.06, respectively) possibly due to low population densities.
* Strathcona County contributes 6.51% to emissions and has a higher average AQHI of 2.34.
* The other two show relatively a moderate level of emission contribution.

**What is the relationship between vehicle registrations, total emissions, and air quality (AQHI) for each CSD?**

In [22]:
df_query_4 = pd.read_sql_query('''
    SELECT 
        CSD,
        SUM(`Vehicle Registration (Count)`) AS total_vehicle_registrations,
        (SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`)) * 100.0 / 
        (SELECT SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`) FROM masterdata) 
        AS total_emissions_percentage,
        AVG(`Adjusted_AQHI`) AS average_aqhi
    FROM 
        masterdata 
    GROUP BY 
        CSD
    ORDER BY 
        total_vehicle_registrations DESC
    LIMIT 5;
''', engine)

df_query_4

CSD  total_vehicle_registrations  total_emissions_percentage  \
0            Calgary                   14789018.0                    3.627634   
1           Edmonton                   10075736.0                    1.525002   
2  Strathcona County                    1368536.0                    6.509499   
3           Red Deer                    1328746.0                    0.025625   
4         Lethbridge                    1204355.0                    0.011308   

   average_aqhi  
0      2.676864  
1      2.685386  
2      2.336999  
3      2.361451  
4      2.382525

* Strathcona County, with significantly fewer vehicle registrations (1.37M), contributes disproportionately high emissions (6.51%).
* Calgary and Edmonton have the highest total vehicle registrations (14.79M and 10.08M, respectively) but relatively low emissions percentages and relatively high average air quality values but still poor. 
* Red Deer and Lethbridge have the fewest vehicle registrations (1.33M and 1.20M, respectively) and negligible emissions contributions.

**How does the number of dwelling units in each CSD compare with the number of vehicle registrations, and does higher vehicle ownership correlate with more emissions?**

In [23]:
df_query_5 = pd.read_sql_query('''
    SELECT 
        CSD,
        SUM(`Dwelling Units (Count)`) AS total_dwelling_units,
        SUM(`Vehicle Registration (Count)`) AS total_vehicle_registrations,
        (SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`)) * 100.0 / 
        (SELECT SUM(`CO2 Equivalent (t)`) + SUM(`N2O Equivalent (t)`) + SUM(`CH4 Equivalent (t)`) FROM masterdata) 
        AS total_emissions_percentage,
        SUM(`Dwelling Units (Count)`) - SUM(`Vehicle Registration (Count)`) AS dwelling_vs_vehicle_diff
    FROM 
        masterdata 
    GROUP BY 
        CSD
    ORDER BY 
        total_vehicle_registrations DESC
    LIMIT 5;
''', engine)

df_query_5

CSD  total_dwelling_units  total_vehicle_registrations  \
0            Calgary            10617526.0                   14789018.0   
1           Edmonton             8056632.0                   10075736.0   
2  Strathcona County              774915.0                    1368536.0   
3           Red Deer              884955.0                    1328746.0   
4         Lethbridge              835318.0                    1204355.0   

   total_emissions_percentage  dwelling_vs_vehicle_diff  
0                    3.627634                -4171492.0  
1                    1.525002                -2019104.0  
2                    6.509499                 -593621.0  
3                    0.025625                 -443791.0  
4                    0.011308                 -369037.0

* In all CSDs listed, the number of vehicle registrations significantly exceeds the number of dwelling units
* Strathcona County shows a strong relationship between higher vehicle ownership and emissions contribution. Despite fewer total vehicles (1.37M) compared to urban centers, its emissions percentage (6.51%) is disproportionately high.
* Urban areas like Calgary and Edmonton have the largest differences, with 4.17M and 2.02M more vehicles than dwellings. This could indicate higher per-household vehicle ownership or a commuter population using vehicles registered in these cities.

* We had missing data for 2001 and 2002. From 2002 onwards, the steady decline in natural gas production does not strongly correlate with AQHI or emissions, indicating other influencing factors such as industrial activity or transportation emissions. The emissions percentage increase suggests environmental impacts are not fully mitigated by reduced production. 